lib/resources/virtualization.rb

```diff
@@ -0,0 +1,275 @@
+# encoding: utf-8
+# author: Takaaki Furukawa
+
+require 'hashie/mash'
+
+module Inspec::Resources
+  class Virtualization < Inspec.resource(1) # rubocop:disable Metrics/ClassLength
+    name 'virtualization'
+    desc 'Use the virtualization InSpec audit resource to test the virtualization platform on which the system is running'
+    example "
+      describe virtualization do
+        its('system') { should eq 'docker' }
+      end
+
+      describe virtualization do
+        its('role') { should eq 'guest' }
+      end
+
+      describe virtualization.systems[:docker] do
+        it { should eq 'guest' }
+      end
+
+      if virtualization.system == 'docker'
```

Comment:
@chris-rock @tkak Thanks this closes a long standing request of mine when we started developing the os profiles. Great work! :) 👍 

Agree, it is also a great way to show an example of good use of `only_if` as it is intended. Can't wait to start using this resource in the OS security profiles.